crypto/x509/x509_req.c

```diff
@@ -286,6 +286,15 @@ void X509_REQ_get0_signature(const X509_REQ *req, const ASN1_BIT_STRING **psig,
         *palg = &req->sig_alg;
 }
 
+void X509_REQ_set0_signature(X509_REQ *req, ASN1_BIT_STRING *psig,
+                             X509_ALGOR *palg)
+{
+    if (psig != NULL)
+        req->signature = psig;
```

Comment:
Shouldn't this free any pre-existing value first?

Reply:
fixed.